api/src/test/java/uk/gov/dhsc/htbhf/claimant/repository/PaymentCycleRepositoryTest.java

```diff
@@ -0,0 +1,163 @@
+package uk.gov.dhsc.htbhf.claimant.repository;
+
+import io.zonky.test.db.AutoConfigureEmbeddedDatabase;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import uk.gov.dhsc.htbhf.claimant.entity.Claim;
+import uk.gov.dhsc.htbhf.claimant.entity.Payment;
+import uk.gov.dhsc.htbhf.claimant.entity.PaymentCycle;
+
+import java.time.LocalDate;
+import java.util.Iterator;
+import java.util.List;
+import javax.validation.ConstraintViolationException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimTestDataFactory.aValidClaim;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.PaymentCycleTestDataFactory.aPaymentCycleWithClaim;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.PaymentCycleTestDataFactory.aPaymentCycleWithPaymentAndClaim;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.PaymentCycleTestDataFactory.aValidPaymentCycleBuilder;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.PaymentTestDataFactory.aPaymentWithClaim;
+
+@SpringBootTest
+@AutoConfigureEmbeddedDatabase
+class PaymentCycleRepositoryTest {
+
+    @Autowired
+    private PaymentCycleRepository paymentCycleRepository;
+
+    @Autowired
+    private ClaimRepository claimRepository;
+
+    @Autowired
+    private PaymentRepository paymentRepository;
+
+    @AfterEach
+    void afterEach() {
+        paymentCycleRepository.deleteAll();
+        paymentRepository.deleteAll();
+        claimRepository.deleteAll();
+    }
+
```

Comment:
Could we have an EnumSource test that checks with claims for each status.

Reply:
Can't do an EnumSource as different statuses produce different results. I have done 2 CsvSources - one with claims that should be included in the results, another with claims that shouldn't.